lib/pangea/pages/class_settings/p_class_widgets/room_capacity_button.dart

```diff
@@ -0,0 +1,152 @@
+import 'package:fluffychat/pages/chat_details/chat_details.dart';
+import 'package:fluffychat/pangea/extensions/pangea_room_extension/pangea_room_extension.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_gen/gen_l10n/l10n.dart';
+import 'package:future_loading_dialog/future_loading_dialog.dart';
+import 'package:matrix/matrix.dart';
+
+class RoomCapacityButton extends StatefulWidget {
+  final Room? room;
+  final ChatDetailsController? controller;
+  const RoomCapacityButton({
+    super.key,
+    this.room,
+    this.controller,
+  });
+
+  @override
+  RoomCapacityButtonState createState() => RoomCapacityButtonState();
+}
+
+class RoomCapacityButtonState extends State<RoomCapacityButton> {
+  Room? room;
+  ChatDetailsController? controller;
+  int? capacity;
+  String? nonAdmins;
+
+  RoomCapacityButtonState({Key? key});
+
+  @override
+  void initState() {
+    super.initState();
+    room = widget.room;
+    controller = widget.controller;
+    capacity = room?.capacity;
+    room?.numNonAdmins.then(
+      (value) => setState(() {
+        nonAdmins = value.toString();
+        overCapacity();
+      }),
+    );
+  }
+
+  Future<void> overCapacity() async {
+    if ((room?.isRoomAdmin ?? false) &&
+        capacity != null &&
+        nonAdmins != null &&
+        int.parse(nonAdmins!) > capacity!) {
+      ScaffoldMessenger.of(context).showSnackBar(
+        SnackBar(
+          content: Text(
+            L10n.of(context)!.roomExceedsCapacity,
+          ),
+        ),
+      );
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final iconColor = Theme.of(context).textTheme.bodyLarge!.color;
+    return Column(
+      children: [
+        ListTile(
+          onTap: () =>
+              ((room?.isRoomAdmin ?? true) ? (setClassCapacity()) : null),
+          leading: CircleAvatar(
+            backgroundColor: Theme.of(context).scaffoldBackgroundColor,
+            foregroundColor: iconColor,
+            child: const Icon(Icons.reduce_capacity),
+          ),
+          subtitle: Text(
+            (capacity == null)
+                ? L10n.of(context)!.capacityNotSet
+                : (nonAdmins != null)
+                    ? '$nonAdmins/$capacity'
+                    : '$capacity',
+          ),
+          title: Text(
+            L10n.of(context)!.roomCapacity,
+            style: TextStyle(
+              color: Theme.of(context).colorScheme.secondary,
+              fontWeight: FontWeight.bold,
+            ),
+          ),
+        ),
+      ],
+    );
+  }
+
+  Future<void> setCapacity(int newCapacity) async {
+    capacity = newCapacity;
+  }
+
+  Future<void> setClassCapacity() async {
```

Comment:
Is this function used to set the capacity for all room (chats and spaces) or just spaces? If it's for all types of rooms, can you rename it "setRoomCapacity"?

Reply:
Also, in this dialog, can you add a subtitle explaining that only non-admins are counted in the room capacity, and also don't allow the user to set a capacity over the room's current participant count?